src/backend/distributed/executor/local_executor.c

```diff
@@ -228,6 +230,17 @@ ExecuteLocalTaskListExtended(List *taskList,
 		EnsureTaskExecutionAllowed(isRemote);
 	}
 
+	/*
+	 * If workerJob has a partitionKeyValue, we need to set the colocation id
+	 * and partition key value for each task before we start executing them
+	 * because tenant stats are collected based on these values of a task.
+	 */
+	if (distributedPlan != NULL && distributedPlan->workerJob != NULL && taskList != NIL)
+	{
+		SetJobColocationId(distributedPlan->workerJob);
+		SetColocationIdAndPartitionKeyValueForTasks(taskList, distributedPlan->workerJob);
```

Comment:
I'm a bit surprised by the fix vs. the PR title. :)

If we set taskList->partitionKeyValue here, does it mean we don't have to set it in other places.

Conversely, if we set it for stored procedures & functions, does it mean we don't have to set it here?

Reply:
You're right that the PR title is not very accurate in describing the changes made. I named it based on the specific use case where I discovered the issue while testing stored procedure calls for tenant stats. I agree that we should have a more descriptive title reflecting the changes made. So I just updated the PR title and description.

Our initial implementation only covered `LocallyExecuteTaskPlan`, but this PR covers `ExecuteLocalTaskListExtended` as well, which is necessary to gather tenant stats properly. This ensures `partitionKeyValue` and `colocationId` are set for tasks in a list if a `workerJob` has these values.

I would like to confirm if there are any other code paths that need to be covered. If you have any suggestions or concerns, please let me know.